### PR TITLE
Add Top N Words Tutorial

### PR DESCRIPTION
**Pull request by amberarrow**
In addition to formatting, I also made minor changes to the text to reduce verbosity, improve clarity and readability.

### CODE REVIEW

File: docs/tutorial-topNwords/TopNWords-c2.md [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+The first customization involves the package name within the files which
+currently reflects the package from which they were copied. The relevant line
+within each file might look like this:
+```
+package com.datatorrent.demos.wordcount;
+```
+Change this line to reflect the current location, for example:
+```
+package com.example.topnwordcount;
+```
```
Comment (sashadt) [inline docs/tutorial-topNwords/TopNWords-c2.md]: This mismatched code block causes the rest of the document to have invalid structure.

Comment (amberarrow) [inline docs/tutorial-topNwords/TopNWords-c2.md]: I don't see what is mismatched there -- both package statements are surrounded by triple back-quotes.
Anyway, feel free to modify as you see fit -- I had a real hard time getting code blocks to look the way I want.
Simple 4-space indents were not having the expected effects in some places though they seem to work in other places.

Comment (sashadt) [inline docs/tutorial-topNwords/TopNWords-c2.md]: Yes, I've ran into that myself.  Markdown requires 4 spaces for regular
code blocks and 8 for comment blocks inside lists.

If you preview the file in GitHub about half way down you can see the
mismatch of the code block.  It causes regular text to be inside the code
block.

Comment (amberarrow) [inline docs/tutorial-topNwords/TopNWords-c2.md]: It looks OK in mkdocs preview.

(Github actually shows an example similar to what I had at:
https://help.github.com/articles/github-flavored-markdown/)

**Pull request merged**